maxim/logger/llamaindex/client.py

```
@@ -0,0 +1,273 @@
+from ...scribe import scribe
+from ..components import Trace, Span, Generation, GenerationConfigDict 
+from .utils import LlamaIndexUtils
+
+_INSTRUMENTED = False
+_global_maxim_trace: contextvars.ContextVar[Union[Trace, None]] = (
+    contextvars.ContextVar("maxim_trace_context_var", default=None)
+)
+_agent_spans: Dict[str, Span] = {}
+_current_generation: Dict[str, Generation] = {}
+
+def instrument_llamaindex(logger: Logger, debug: bool = False):
```

Ruff (0.12.2)

`instrument_llamaindex` is too complex (36 > 10)

(C901)